articles/getting-started-with-jwt-using-angular8-and-nodejs/index.md

```diff
@@ -0,0 +1,293 @@
+
+### Creating the Angular App
+Create a base folder and navigate to the folder. Then, using a terminal run the following command:
+```bash
+$ ng new angFrontend
+```
+The command will create an Angular app in your base folder with a new subfolder named the same as the app's name. In this case, `angFrontend`. It comes with preinstalled libraries required by a typical Angular app.
+
+### Creating an Interceptor
+Next, we will be creating the interceptor. First, browse to the folder where the angular app was created and run the following command to achieve that:
+```bash
+$ ng generate service AuthInterceptor
+```
+Then, navigate to `src/app` and edit the file `app.module.ts` to look like the code below. I import the HTTP Module for HTTP calls. I also make the interceptor that has just been created to be a provider and so that it will have global access to all HTTP calls.
+```typescript
+//the modules required by the app are imported here
+import { BrowserModule } from '@angular/platform-browser'; // this ensures the application will run on the browser
+import { NgModule } from '@angular/core';
+
+import { AppRoutingModule } from './app-routing.module'; //ensures the application have routing capabilities
+
+import { AppComponent } from './app.component'; //made present for bootstrapping application on the launch
+
+import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http';//enables the application to communicate with the backend services
+import { AuthInterceptorService } from './auth-interceptor.service';//this will allow the app to automatically attach authorization information to requests
+import { HomeComponent } from './home/home.component';//implements the home route
+
+@NgModule({
+  declarations: [
+    AppComponent,
+    HomeComponent
+  ],
+  imports: [
+    BrowserModule,
+    AppRoutingModule,
+    
+    HttpClientModule
+  ],
+  providers: [
+    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptorService, multi: true }
+  ],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+
+Next, edit `src/app/auth-interceptor.service.ts` to look like shown below:
+
+```typescript
+import { Injectable } from '@angular/core';//imports the class that provides local storage for token
+import { HttpEvent, HttpInterceptor, HttpHandler, HttpRequest, HttpErrorResponse } from '@angular/common/http';
+import { catchError, filter, take, switchMap } from "rxjs/operators";
+import { Observable, throwError } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class AuthInterceptorService implements HttpInterceptor {
+
+  intercept(req: HttpRequest<any>, next: HttpHandler) {
+    console.log("Interception In Progress"); //Interception Stage
+    const token: string = localStorage.getItem('token');//This retrieves a token from local storage
+    req = req.clone({ headers: req.headers.set('Authorization', 'Bearer ' + token) });//This clones HttpRequest and Authorization header with Bearer token added
```

Comment:
You have included the code to send the JWT but none is sent since you haven't imported this service in your home component. Please update your code and add the missing code.